### PR TITLE
Add area picker

### LINKED ISSUE #4596: Allow creating an area from the area dropdown picker
## The request

Area dropdowns should end with an option "Add new area…". Clicking that will prompt for area name. Once entered, it will be added to dropdown and selected.

We should use this dropdown in all places where we allow picking areas (config flow finish, device edit modal). In config flow finish, we can remove the "add area" button.

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

image

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/home-assistant-polymer/issues/4596
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
One issue:
Config entry dialog is too small to display 2 area pickers next to each other, make the dialog wider, or move each device to own line? Or change the area picker?
image

File: src/components/ha-area-picker.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  }
+
+  private _openedChanged(ev: PolymerChangedEvent<boolean>) {
+    this._opened = ev.detail.value;
+  }
+
+  private _areaChanged(ev: PolymerChangedEvent<string>) {
+    const newValue = ev.detail.value;
+
+    if (newValue === "add_new") {
```
Comment (balloob) [inline src/components/ha-area-picker.ts]: FYI, if you have a large IF-block at which you return at the end, and then have a small if-block, you can better reverse them and then have less indentation. Easier to follow.

```ts
    if (newValue !== this._value) {
      this._setValue(newValue);
    }

    if (newValue !== "add_new") { return; }
```

File: src/dialogs/config-flow/dialog-data-entry-flow.ts [hunk 2]
```diff
@@ -322,7 +322,7 @@ class DataEntryFlowDialog extends LitElement {
       haStyleDialog,
       css`
         ha-paper-dialog {
-          max-width: 500px;
+          max-width: 600px;
```
Comment (balloob) [inline src/dialogs/config-flow/dialog-data-entry-flow.ts]: I know why you did this, but I think for a future PR we should just convert the grid to a single list. Left side name/manufactuer/model, right side area. I'll open an issue for it.

**Review by balloob — APPROVED** [review]
Awesome !

**Pull request merged**